fuel/utils/write_lock.py

```diff
@@ -0,0 +1,327 @@
+import atexit
```

Comment:
I checked it and I haven't found libraries for that. Stackoverflow mostly suggests to copy-paste a piece of code of questionable quality that doesn't cover all cases. Should we create a small library?